src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -502,7 +502,7 @@ protected function initializeBundles()
      */
     protected function getContainerClass()
     {
-        return $this->name.ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
+        return preg_replace('/(^[0-9]*|[^a-zA-Z0-9_]+)/', '', $this->name.ucfirst($this->environment)).($this->debug ? 'Debug' : '').'ProjectContainer';
```

Comment:
I don't think we need to support these edge cases... especially now that Melody has been fixed. it adds a regex for every single request and makes the code less readable.